airflow/models.py

```diff
@@ -2777,3 +2766,150 @@ class ImportError(Base):
     timestamp = Column(DateTime)
     filename = Column(String(1024))
     stacktrace = Column(Text)
+
+
+class Trigger:
+    """
+    Triggers represent a dependency of an operator on a range of schedules of
+    another operator.
+    * Trigger(t) depends on the state of t of the same schedule
+    * Trigger(t, 1) depends on the state of t of the previous schedule
+    * Trigger(t, n) depends on the state of t of the n previous schedules
+      (not including the current schedule)
+
+    :param task: the depended upon task
+    :type task: TaskInstance
+    :param past_executions: The number of past executions that the
+    dependant depends on. If it is left to 0, the Trigger is aimed at
+    the same schedule as the dependant. If it is set to a value > 0,
+    the Trigger will look at the corresponding past executions to
+    evaluate the Trigger.
+    :type past_executions: non negative integer
+    """
+
+    def __init__(self, task, past_executions=0):
+        self.dag_id = task.dag_id
+        self.task_id = task.task_id
+        self.task = task
+
+        assert past_executions >= 0, "past_executions parameter must be >= 0"
+        self.past_executions = past_executions
+
+    def states(self, execution_date, main_session=None):
```

Comment:
This method returns a list of states, which is enough for its current purpose (i.e. being called from evaluate), but we lose the information of which states corresponds to which execution date, even though we have it during the execution of the method

=> I'd rather return a dict (date -> state) from here and, in `evaluate`, discard the keys since they are not needed there.